packages/core/contracts/oracle/implementation/OptimisticOracle.sol

```diff
@@ -285,6 +305,7 @@ contract OptimisticOracle is OptimisticOracleInterface, Testable, Lockable {
             "proposePriceFor: Requested"
         );
         Request storage request = _getRequest(requester, identifier, timestamp, ancillaryData);
+        if (request.eventBased) require(proposedPrice != type(int256).min, "Cannot propose 'too early'");
```

Comment:
This now puts the magic number into the smart contract which is kind of a big deal since we can no longer change the magic number off-chain. This isn't a huge deal, but makes me wonder: why is this check that proposedPrice != "too early" even important? Can't we just let disputers/UMA voters just reject any magic number price for an event-based request?

Reply:
It's a great question.

I think it's important because of the following situation.

1. Requesting contract requests the price on startup. This requests includes a reward for a correct response.
2. Instead of waiting for the event to occur, a clever proposer could just propose the "too early" value immediately. This doesn't actual provide anything valuable that's worthy of the reward. However, this is technically correct since the DVM would also indicate that this price request was submitted too early, making any disputer lose money for disputing it.
3. The proposer is, therefore, not disputed, and receives the reward. The requesting contract would need to resubmit a new request to get the actual value after the event occurs.

Reply:
Could this instead be enforced by the requesting contract using the proposePrice callback?

Reply:
it is already put into the contracts in other places. see: https://github.com/UMAprotocol/protocol/blob/6368e90125c3b1ae445ff0ae3c3b38b63efaa72a/packages/core/contracts/financial-templates/long-short-pair/LongShortPair.sol#L348 where we do this in the LPS (the origin of the "magic number" was this contract in this place).

I personally think this is fine and is the standard already converged on in at least two places (including external inergrations). the reasoning for using `type(int256).min` around being a very unlikely price, in fact the most unlikely price, still holds true.

Reply:
Ok good point @chrismaree good to leave as is